lib/streamlit/commands/page_config.py

```diff
@@ -142,3 +179,14 @@ def get_random_emoji():
     # Weigh our emojis 10x, cuz we're awesome!
     # TODO: fix the random seed with a hash of the user's app code, for stability?
     return random.choice(RANDOM_EMOJIS + 10 * ENG_EMOJIS)
+
+
+def fix_url(url):
+    p = urllib.parse.urlparse(url, "http")
+    netloc = p.netloc or p.path
+    path = p.path if p.netloc else ""
+    if not netloc.startswith("www."):
+        netloc = "www." + netloc
+
+    p = urllib.parse.ParseResult("http", netloc, path, *p[3:])
+    return p.geturl()
```

Comment:
Would be good to add a couple tests for this. Also, I believe this code was taken from stackoverflow, so it'd be good to leave a comment/docstring mentioning the source